src/frontend/src/catalog.rs

```diff
@@ -82,7 +82,12 @@ impl CatalogManager for FrontendCatalogManager {
         Ok(true)
     }
 
-    async fn deregister_table(&self, _request: DeregisterTableRequest) -> CatalogResult<bool> {
+    async fn deregister_table(&self, request: DeregisterTableRequest) -> CatalogResult<bool> {
+        let table_name = TableName::new(request.catalog, request.schema, request.table_name);
+        self.partition_manager
+            .table_routes()
+            .invalid_table_route(&table_name)
```

Comment:
This function name should be renamed to `invalidate_table_route`. `invalid` is not a verb.